src/utils.ts

```diff
@@ -6,6 +6,52 @@
  */
 import { ContextAttributes } from './constructor-types';
 
+function chceksum(obj: any): string {
+  const str = JSON.stringify(obj);
+  let hash = 0;
+  for (let i = 0; i < str.length; i++) {
+    const chr = str.charCodeAt(i);
+    hash = (hash << 5) - hash + chr;
+    hash |= 0;
+  }
+  return hash.toString(16);
+}
+
+/**
+ * a class that hold an array of objects that each object would be unique
+ *
+ */
+class ObjectSet {
+  dataArray: any[] = [];
+  dataDict: any = {};
```

Comment:
If it is always an object then you could give it a type of Record<string, unknown>, Record basically creates an object type with key type of string, and value type of unknown because you really dont know what the values of this object will be.